pkg/datapath/loader/base.go

```diff
@@ -217,6 +217,16 @@ func (l *Loader) reinitializeIPSec(ctx context.Context) error {
 	return nil
 }
 
+func (l *Loader) ReinitializeXDP(ctx context.Context, extraCArgs []string) error {
```

Comment:
- This method can be unexported, it's only used in `(*Loader).Reinitialize()` below.
- As discussed offline, a comment describing how and why to use this would be welcome. :+1: 
- Since this method only really takes strings, what about making it variadic?
```suggestion
func (l *Loader) ReinitializeXDP(ctx context.Context, extraCArgs ...string) error {
```
Then it can later be called as
```go
l.ReinitializeXDP(ctx, "-Dcapture_enabled=0")
```
instead of
```go
l.ReinitializeXDP(ctx, []string{"-Dcapture_enabled=0"})
```

Reply:
Thanks! I added comments to exposed functions. The ReinitializeXDP() cannot be unexported given we use it from the recorder right now. Re variadic, no strong opinion from my side, it feels potentially a bit more complex given we pass in different number of extra args from the recorder side into a single ReinitializeXDP. I presume to make that work we would need to split the code have have two separate calls into ReinitializeXDP.. don't think it's really making it much better tbh.